File: Observer/c#/StatisticsDisplay.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Observer
{
    public class StatisticsDisplay: IDisplayElement, IObserver
    {
        private List<float> temperatureHistory;
        private ISubject _weatherData;

        public StatisticsDisplay(ISubject weatherData)
        {
          temperatureHistory = new List<float>();
          _weatherData = weatherData;
          _weatherData.registerObserver(this);
        }

        public void display()
        {
          System.Console.WriteLine($@"Avg/Max/Min temperature = {
            temperatureHistory.Average().ToString("n2") }/{
            temperatureHistory.Max().ToString("n2") }/{
            temperatureHistory.Min().ToString("n2") }");
        }

        public void update(float temperature, float humidity, float pressure)
        {
          temperatureHistory.Add(temperature);
          display();
        }
    }
}

File: Observer/typescript/src/StatisticsDisplay.ts
import IDisplayElement from "./IDisplayElement";
import IObserver from "./IObserver";
import ISubject from "./ISubject";

export class StatisticsDisplay implements IDisplayElement, IObserver {
  private temperatureHistory: number[] = [];
  private _weatherData: ISubject;

  constructor(weatherData: ISubject) {
    this._weatherData = weatherData;
    this._weatherData.registerObserver(this);
  }

  display() {
    console.log(`Avg/Max/Min temperature = ${
      this.getAverage().toFixed(2)}/${
      this.getMax().toFixed(2)}/${
      this.getMin().toFixed(2)}`);
  }

  update(temperature: number) {
    this.temperatureHistory.push(temperature);
    this.display();
  }

  getAverage() {
    const sum = this.temperatureHistory.reduce((previous, current) => previous + current);
    return sum / this.temperatureHistory.length;
  }

  getMin() {
    return this.temperatureHistory.reduce((previous, current) => previous > current ? current : previous,
    this.temperatureHistory[0]);
  }

  getMax() {
    return this.temperatureHistory.reduce((previous, current) => previous < current ? current : previous,
    this.temperatureHistory[0]);
  }
}